types/mongodb/index.d.ts

```diff
@@ -1204,7 +1204,7 @@ export type PushOperator<TSchema> = ({
 export type PullOperator<TSchema> = ({
     readonly [key in KeysOfAType<TSchema, any[]>]?: Unpacked<TSchema[key]> | QuerySelector<Unpacked<TSchema[key]>>;
 } &
-    NotAcceptedFields<TSchema, any[]>) & {
+    NotAcceptedFields<TSchema, any[]>) | {
```

Comment:
I think you have just opened the types to accept anything and reverted the previous typings.
Please add some failing tests to updateX files. e.g. you should not accept string in a number field. Checkout ‘filterQuery’ file on how to write failing tests.

Reply:
I see.  The tests I added are failing tests if you revert this simple commit.  I loosened them to at least have support for them.  Some support is better than no support?  I'd be more pro loosen then figure out how to actually make them strict and work while allowing all tests to pass.  Currently I have to @ts-ignore everything or cast stuff as "any".

Reply:
@CaselIT I was actually pretty much on this same track earlier but didn't want to spend too much time.